python/ray/autoscaler/_private/aws/config.py

```diff
@@ -511,6 +522,16 @@ def _configure_subnet(config):
             subnet_src_info[key] = "default"
             node_config["SubnetIds"] = subnet_ids
         else:
+            node_type_subnets = node_config["SubnetIds"]
+            if not _check_subnets_match_azs(ec2, azs, subnet_ids):
```

Comment:
> However, while this may correctly ensure that all user-specified Subnet IDs belong to the provided Availability zones, some of these subnets may still be unavailable, or may not belong to the VPC of the security group.

By `VPC of the security group.` do you mean the head subnet & the head's SG should be the same VPC?